dataprofiler/profilers/profiler_options.py

```diff
@@ -416,8 +421,17 @@ def _validate_helper(self, variable_path='FloatOptions'):
         errors = super()._validate_helper(variable_path=variable_path)
         if not isinstance(self.precision, BooleanOption):
             errors.append("{}.precision must be a BooleanOption."
+                          .format(variable_path))        
+        errors += self.precision._validate_helper(
+            variable_path + '.precision')
+        
+        
+        if self.precision_sample_ratio is not None \
+           and isinstance(self.precision_sample_ratio, float) \
+           and self.precision_sample_ratio < 0 and self.precision_sample_ratio > 1.0:
+            errors.append("{}.precision_sample_ratio must be a float between 0 and 1."
```

Comment:
Should have error get raised if precision sample ratio isn't float at all (what if they passed a string or boolean)? In that case this if condition won't pick it up and add the error. You could do this via 
`if <not None> and (<not float> or <not btwn 0,1>)`